File: Java/CheckOutApp.java
import java.util.Scanner;

public class CheckOutApp {
	public static void main(String[] args) {
	Scanner input = new Scanner(System.in);

	System.out.println("\nWELCOME TO SEMICOLON STORE..");

	System.out.println("\nWhat is the customer's Name?");
	String customerName = input.nextLine();

	String[] itemBought = new String[10];
	double[] pricePerUnit = new double[10];
	int[] quantities = new int[10];
	int itemNumber = 0;

	String choice = "yes";

	do {
		System.out.print("\nWhat did the customer buy? ");
		itemBought[itemNumber] = input.next();
    
		System.out.print("\nHow much per unit? ");
		double price = input.nextDouble();
		if (price <= 0) {
		System.out.println("Enter a valid price");
		continue;
}	
		pricePerUnit[itemNumber] = price;
		
		System.out.print("\nHow many unit? ");
		int quantity = input.nextInt();
		if (quantity <= 0) {
		System.out.println("Invalid number of units!");
		continue;
}
		quantities[itemNumber] = quantity;
		
		itemNumber++;
		
		System.out.println("Add more items? (yes or no)");
		choice = input.next();
		if (!choice.equalsIgnoreCase("yes") && !choice.equalsIgnoreCase("no")) {
		System.out.println("Please enter yes or no.");
} 
} 	while (!choice.equalsIgnoreCase("no"));

		System.out.println("Enter cashier's Name?");
		String cashierName = input.next();

		System.out.println("How much discount will customer get?");
		double discount = input.nextDouble();
		if (discount < 0 || discount > 100) {
		System.out.println("Discount must be between 0 and 100!");
}
		System.out.println(CheckOutAppFunctions.getStoreDetails());
		CheckOutAppFunctions.printCustomerInvoice(itemBought, quantities, pricePerUnit, itemNumber, customerName, cashierName, discount);

		System.out.println("\n\nHow much did the customer give to you?");
		double payment = input.nextDouble();
		if (payment <= 0) {
		System.out.println("Invalid or negative payment!");
}
		System.out.println(CheckOutAppFunctions.getStoreDetails());
		CheckOutAppFunctions.printCustomerReceipt(itemBought, quantities, pricePerUnit, itemNumber, customerName, cashierName, discount, payment);
}
}

File: Python/check_out_app.py
from datetime import datetime

class check_out_app:

	def get_store_details():
		return "SEMICOLON STORES\nMAIN BRANCH\nLOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS.\nTEL: 03938373320\n" + str(datetime.now())

	def printCustomerInvoice(items, quantities, prices, CustomerName, cashierName, discount):

		print("Cashier: " + cashierName)
		print("Customer Name: " + CustomerName)

		for line in range (1, 61):
			print("=", end = " ")

		print("\nITEM\t\tQUANTITY\tPRICE\tTOTAL(NGN)")
		for dash in range (1, 61):
			print("-", end = " ")

		totalItems = prices * quantities

		print(f"\n{items}\t\t{quantities}\t{price}\t{totalItems}")

		for dash2 in range (1, 61):
			print("-", end = " ")

		discountAmount = total * (discount / 100)
		finalAmount = total - discountAmount
		VAT = total * (17.50 / 100)
		billTotal = VAT + total - discountAmount

		print("Sub Total: " + total)
		print("Discount: " + discountAmount)
		print("VAT @ 17.50%: " + VAT)

		for line2 in range (1, 61):
			print("=", end = " ")

		print("Bill Total: " + billTotal)

		for line3 in range (1, 61):
			print("=", end = " ")

		print("THIS IS NOT A RECEIPT KINDLY PAY " + billTotal)

		for line4 in range (1, 61):
			print("=", end = " ")

	def printCustomerInvoice(items, quantities, prices, CustomerName, cashierName, discount, payment):
		print("Cashier: " + cashierName)
		print("Customer Name: " + CustomerName)

		for line in range (1, 61):
			print("=", end = " ")

		print("\nITEM\t\tQUANTITY\tPRICE\tTOTAL(NGN)")
	
		for dash in range (1, 61):
			print("-", end = " ")

		totalItems = prices * quantities
	
		print(f"\n{items}\t\t{quantities}\t{price}\t{totalItems}")

		for dash2 in range (1, 61):
			print("-", end = " ")

		discountAmount = total * (discount / 100)
		finalAmount = total - discountAmount
		VAT = total * (17.50 / 100)
		billTotal = VAT + total - discountAmount
		balance = payment - billTotal

		print("Sub Total: " + total)
		print("Discount: " + discountAmount)
		print("VAT @ 17.50%: " + VAT)

		for line2 in range (1, 61):
			print("=", end = " ")

		print("Bill Total: " + billTotal)
		print("Amount paid: " + payment)
		print("Balance: " + balance)

		for line3 in range (1, 61):
			print("=", end = " ")
		pint("THANK YOU FOR YOUR PATRONAGE")

		for line4 in range (1, 61):
			print("=", end = " ")

def main():

	while True:
		name = input("What is the customer's Name? ")
		itemBought
		icePerUnit
cashierName;
discount;
payment;
quantities = 0;


	do {
	System.out.print("What did the customer buy? ");
	itemBought = input.next();
	
	System.out.print("How much per unit? ");
	pricePerUnit = input.nextDouble();
	if (pricePerUnit <= 0) {
	System.out.println("Enter a valid price");
	continue;
}
	System.out.println("Add more items?");
	choice = input.next();
	if (!choice.equalsIgnoreCase("yes") && !choice.equalsIgnoreCase("no")) {
	System.out.println("Please enter 'yes' or 'no'.");
} else { quantities += 1; }
}
	while (!choice.equalsIgnoreCase("no"));

	System.out.println("What is your Name?");
	cashierName = input.next();

	System.out.println("How much discount will customer get?");
	discount = input.nextDouble();

	if (discount < 0 || discount > 100) {
	System.out.println("Discount must be between 0 and 100!");
	continue;
} else {
	System.out.println("How much did the customer give to you?");
	payment = input.nextDouble();
	if (payment <= 0) {
	System.out.println("Invalid or negative payment!");
}
}
}
	System.out.println(CheckOutAppFunctions.getStoreDetails());
	CheckOutAppFunctions.printCustomerInvoice(itemBought, quantities, pricePerUnit, customerName, cashierName, discount);

}
}
